webapp/packages/plugin-codemirror6/src/useEditorDefaultExtensions.ts

```diff
@@ -36,53 +36,66 @@ DEFAULT_KEY_MAP.push({
 
 export interface IDefaultExtensions {
   lineNumbers?: boolean;
+  tooltips?: boolean;
+  highlightSpecialChars?: boolean;
+  syntaxHighlighting?: boolean;
+  bracketMatching?: boolean;
+  dropCursor?: boolean;
+  crosshairCursor?: boolean;
+  foldGutter?: boolean;
+  highlightActiveLineGutter?: boolean;
+  highlightActiveLine?: boolean;
+  indentOnInput?: boolean;
+  rectangularSelection?: boolean;
+  keymap?: boolean;
 }
 
-const DEFAULT_EXTENSIONS_COMPARTMENT = new Compartment();
-
-/** Provides the necessary extensions to establish a basic editor */
-export function useEditorDefaultExtensions(options?: IDefaultExtensions): [Compartment, Extension] {
-  return useMemo(() => {
-    const extensions = [];
-    if (options?.lineNumbers) {
-      extensions.push(lineNumbers());
-    }
+const extensionMap = {
+  lineNumbers,
+  tooltips: () => tooltips({ parent: document.body }),
+  highlightSpecialChars,
+  syntaxHighlighting: () => syntaxHighlighting(classHighlighter),
+  bracketMatching,
+  dropCursor,
+  crosshairCursor,
+  foldGutter: () =>
+    foldGutter({
+      markerDOM: (open: boolean) => {
+        const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+        svg.setAttributeNS(null, 'viewBox', '0 0 15 8');
+        svg.style.maxWidth = '100%';
+        svg.style.maxHeight = '100%';
```

Comment:
You need to initialize it in a memoized function (these extensions must be independent instances for every editor)